File: business-easy-core/src/main/java/com/businesseasy/core/entities/DocumentEntity.java
package com.businesseasy.core.entities;

import com.businesseasy.core.common.enums.DocumentType;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "document")
public class DocumentEntity extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "people_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private PeopleEntity people;

    @Column(name="type")
    @Enumerated(value = EnumType.STRING)
    private DocumentType type;

    @Column(name = "total")
    private BigDecimal total;

    @Column(name = "profit")
    private BigDecimal profit;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "document_id", referencedColumnName = "id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private List<DocumentItemEntity> documentItems;

}

File: easiness-ui/app/server/entity/document.entity.ts
import Big from "big.js";
import { Column, Entity, ManyToOne, OneToMany, PrimaryGeneratedColumn } from "typeorm";
import { BigColumn } from "../utils/decorators";
import { Base } from "./base.entity";
import { DocumentItem } from "./document-item.entity";
import { People } from "./people.entity";
import { Tx } from "./tx.entity";

export enum DocumentType {
    PURCHASE_ORDER = "PURCHASE_ORDER",
    INVOICE = "INVOICE"
}

@Entity({ name: 'document' })
export class Document extends Base {
    @PrimaryGeneratedColumn()
    id: number

    @ManyToOne(() => People)
    people: People
    @Column()
    peopleId: number

    @Column()
    type: DocumentType

    @BigColumn()
    total: Big

    @BigColumn(true)
    profit!: Big

    @OneToMany(() => DocumentItem, (item) => item.document, { cascade: true })
    items: DocumentItem[]

    @OneToMany(() => Tx, (item) => item.document)
    payments: Tx[]

}